app/models/solidus_paypal_commerce_platform/gateway.rb

```diff
@@ -23,15 +23,19 @@ def purchase(money, source, options)
       request = OrdersCaptureRequest.new(source.paypal_order_id)
       request.request_body(amount: { currency_code: options[:currency], value: Money.new(money).dollars })
       response = @client.execute_with_response(request)
-      capture_id = response.params["result"].purchase_units[0].payments.captures[0].id
-      source.update(capture_id: capture_id) if response.success?
+      if response.success?
+        capture_id = response.params["result"].purchase_units[0].payments.captures[0].id
+        source.update(capture_id: capture_id) if response.success?
```

Comment:
The tail check for `if response.success?` has been left there during the refactoring ✂️

Reply:
Whoops! 😬